homeassistant/components/mqtt/vacuum/__init__.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for MQTT vacuums.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.mqtt/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.vacuum import DOMAIN
+from homeassistant.components.mqtt import ATTR_DISCOVERY_HASH
+from homeassistant.components.mqtt.discovery import (
+    MQTT_DISCOVERY_NEW, clear_discovery_hash)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['mqtt']
+
+CONF_SCHEMA = 'schema'
```

Comment:
Maybe this could instead be called `component`, with options `legacy` or `statevacuum`?